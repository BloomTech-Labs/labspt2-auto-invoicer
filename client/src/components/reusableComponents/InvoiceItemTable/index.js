import React, { Component } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableHead from "@material-ui/core/TableHead";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";

import TextField from "@material-ui/core/TextField";

const styles = theme => ({
  root: {
    width: "100%",
    marginTop: theme.spacing.unit * 3,
    overflowX: "auto"
  },
  table: {
    minWidth: 700
  }
});

class index extends Component {
  constructor() {
    super();
    this.state = {
      itemDescription: "",
      itemQuantity: "",
      itemRate: "",
      itemAmount: "",
      subtotal: "",
      discount: "",
      tax: "",
      shipping: "",
      total: "",
      amountPaid: ""
    };
  }

  changeHandler = event => {
    this.setState({
      [event.target.name]: event.target.value
    });
  };

  onSubmit = event => {
    event.preventDefault();
    this.props.onSubmit2(this.state);
  };

  render() {
    return (
      <div>
        <Paper>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell>Item</TableCell>
                <TableCell align="right">Quantity</TableCell>
                <TableCell align="right">Rate</TableCell>
                <TableCell align="right">Amount</TableCell>
              </TableRow>
            </TableHead>
            <TableBody>
              <TableRow>
                <TableCell>
                  <TextField
                    id="itemDescription"
                    name="itemDescription"
                    placeholder="Item Description"
                    value={this.state.itemDescription}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
                <TableCell align="right">
                  <TextField
                    id="itemQuantity"
                    name="itemQuantity"
                    placeholder="Item Quantity"
                    value={this.state.itemQuantity}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
                <TableCell align="right">
                  <TextField
                    id="itemRate"
                    name="itemRate"
                    placeholder="Rate"
                    value={this.state.itemRate}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
                <TableCell align="right">
                  <TextField
                    id="itemAmount"
                    name="itemAmount"
                    placeholder="Amount"
                    value={this.state.itemAmount}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell rowSpan={6} />
                <TableCell colSpan={2}>Subtotal</TableCell>
                <TableCell align="right">
                  <TextField
                    id="subtotal"
                    name="subtotal"
                    placeholder="Subtotal"
                    value={this.state.subtotal}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell colSpan={2}>Discount</TableCell>
                <TableCell align="right">
                  <TextField
                    id="discount"
                    name="discount"
                    placeholder="Discount"
                    value={this.state.discount}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell colSpan={2}>Tax</TableCell>

                <TableCell align="right">
                  <TextField
                    id="tax"
                    name="tax"
                    placeholder="Tax"
                    value={this.state.tax}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell colSpan={2}>Shipping</TableCell>
                <TableCell align="right">
                  <TextField
                    id="shipping"
                    name="shipping"
                    placeholder="Shipping"
                    value={this.state.shipping}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell colSpan={2}>Total</TableCell>
                <TableCell align="right">
                  <TextField
                    id="total"
                    name="total"
                    placeholder="Total"
                    value={this.state.total}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>

              <TableRow>
                <TableCell colSpan={2}>Amount Paid:</TableCell>
                <TableCell align="right">
                  <TextField
                    id="amountPaid"
                    name="amountPaid"
                    placeholder="Amount Paid"
                    value={this.state.amountPaid}
                    onChange={event => this.changeHandler(event)}
                  />
                </TableCell>
              </TableRow>
            </TableBody>
          </Table>
          <button onClick={event => this.onSubmit(event)}>Submit</button>
        </Paper>
      </div>
    );
  }
}

index.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(index);
