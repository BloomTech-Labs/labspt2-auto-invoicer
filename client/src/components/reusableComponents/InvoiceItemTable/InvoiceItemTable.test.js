import React from "react";
import { render } from "react-testing-library";
import "jest-dom/extend-expect";
import InvoiceItemTable from "../InvoiceItemTable";

describe("<InvoiceItemTable />", () => {
  it("renders Invoice Item Table", () => {
    const { asFragment } = render(<InvoiceItemTable />);
    expect(asFragment()).toMatchSnapshot();
  });
});
