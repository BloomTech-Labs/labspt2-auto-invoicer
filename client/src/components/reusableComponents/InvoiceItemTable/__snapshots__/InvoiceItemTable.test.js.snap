// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InvoiceItemTable /> renders Invoice Item Table 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root-3 MuiPaper-elevation2-7 MuiPaper-rounded-4"
    >
      <table
        class="MuiTable-root-30"
      >
        <thead
          class="MuiTableHead-root-31"
        >
          <tr
            class="MuiTableRow-root-32 MuiTableRow-head-35"
          >
            <th
              class="MuiTableCell-root-37 MuiTableCell-head-38"
              scope="col"
            >
              Item
            </th>
            <th
              class="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
              scope="col"
            >
              Quantity
            </th>
            <th
              class="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
              scope="col"
            >
              Rate
            </th>
            <th
              class="MuiTableCell-root-37 MuiTableCell-head-38 MuiTableCell-alignRight-47"
              scope="col"
            >
              Amount
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root-49"
        >
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="itemDescription"
                    name="itemDescription"
                    placeholder="Item Description"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="itemQuantity"
                    name="itemQuantity"
                    placeholder="Item Quantity"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="itemRate"
                    name="itemRate"
                    placeholder="Rate"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="itemAmount"
                    name="itemAmount"
                    placeholder="Amount"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              rowspan="6"
            />
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Subtotal
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="subtotal"
                    name="subtotal"
                    placeholder="Subtotal"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Discount
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="discount"
                    name="discount"
                    placeholder="Discount"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Tax
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="tax"
                    name="tax"
                    placeholder="Tax"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Shipping
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="shipping"
                    name="shipping"
                    placeholder="Shipping"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Total
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="total"
                    name="total"
                    placeholder="Total"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-32"
          >
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39"
              colspan="2"
            >
              Amount Paid:
            </td>
            <td
              class="MuiTableCell-root-37 MuiTableCell-body-39 MuiTableCell-alignRight-47"
            >
              <div
                class="MuiFormControl-root-50"
              >
                <div
                  class="MuiInputBase-root-67 MuiInput-root-54 MuiInput-underline-58 MuiInputBase-formControl-68 MuiInput-formControl-55"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input-77 MuiInput-input-62"
                    id="amountPaid"
                    name="amountPaid"
                    placeholder="Amount Paid"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
      <button>
        Submit
      </button>
    </div>
  </div>
</DocumentFragment>
`;
